lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -768,6 +819,99 @@ UsdTimeCode MayaUsdProxyShapeBase::GetOutputTime(MDataBlock dataBlock) const
     return _GetTime(dataBlock);
 }
 
+void MayaUsdProxyShapeBase::copyInternalData(MPxNode* srcNode)
+{
+    MStatus retValue { MS::kSuccess };
+
+    // get the srource data block
+    MayaUsdProxyShapeBase* srcProxyShapeBase = static_cast<MayaUsdProxyShapeBase*>(srcNode);
+    MDataBlock             srcDataBlock = srcProxyShapeBase->forceCache();
+
+    // ---------------------------------------------------------------------------------
+    // copyInternalData is called multiple times so we do have to protect against it.
+    // ---------------------------------------------------------------------------------
+
+    // first, read the input value from "outStageDataAttr". outStageDataAttr gets computed when
+    // we get the stage on the proxy. If there is no incoming data, we return right away.
+    MDataHandle srcInDataCachedHandle = srcDataBlock.inputValue(outStageDataAttr, &retValue);
+    if (srcInDataCachedHandle.data().isNull()) {
+        return;
+    }
+
+    // query form the destination block to make sure inStageDataCachedAttr attribute is clean. If it
+    // is clean that means we already have the attr value.
+    MDataBlock dataBlock = forceCache();
+    if (dataBlock.isClean(inStageDataCachedAttr)) {
+        return;
+    }
+
+    // get the handle inDataCachedHandle and return if it doesn't have the data.
+    MDataHandle inDataCachedHandle = dataBlock.outputValue(inStageDataCachedAttr, &retValue);
+    if (inDataCachedHandle.data().isNull()) {
+        return;
+    }
+
+    MayaUsdStageData* srcInData
+        = dynamic_cast<MayaUsdStageData*>(srcInDataCachedHandle.asPluginData());
+    if (!srcInData || !srcInData->stage) {
+        return;
+    }
+
+    // get the pointer to source stage
+    UsdStageRefPtr srcUsdStage = srcInData->stage;
+
+    // transfer session layer
+    // session layer is never shared so transfer its content always.
+    SdfLayerRefPtr sessionLayer
+        = SdfLayer::CreateAnonymous(kAnonymousLayerName + kSessionLayerPostfix + ".usda");
+    sessionLayer->TransferContent(srcUsdStage->GetSessionLayer());
+
+    // decide if the root layer needs to be shared or deep copied.
+    SdfLayerRefPtr rootLayer = SdfLayer::CreateAnonymous(kAnonymousLayerName);
```

Comment:
Shouldn't we move the CreateAnonymous() call inside the conditional, just before line 872?